Xamarin.Essentials/Share/Share.mac.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AppKit;
+using Foundation;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Share
+    {
+        static Task PlatformRequestAsync(ShareTextRequest request)
+        {
+            var items = new List<NSObject>();
+            if (!string.IsNullOrWhiteSpace(request.Title))
+                items.Add(new NSString(request.Title));
+            if (!string.IsNullOrWhiteSpace(request.Text))
+                items.Add(new NSString(request.Text));
+            if (!string.IsNullOrWhiteSpace(request.Uri))
+                items.Add(NSUrl.FromString(request.Uri));
+
+            return PlatformShowRequestAsync(request, items);
+        }
+
+        static Task PlatformRequestAsync(ShareFileRequest request)
+        {
+            var items = new List<NSObject>();
+            if (!string.IsNullOrWhiteSpace(request.Title))
+                items.Add(new NSString(request.Title));
+            if (request.File != null)
+                items.Add(NSUrl.FromFilename(request.File.FullPath));
+
+            return PlatformShowRequestAsync(request, items);
+        }
+
+        static Task PlatformShowRequestAsync(ShareRequestBase request, List<NSObject> items)
+        {
+            var window = NSApplication.SharedApplication.MainWindow;
+            var view = window.ContentView;
+
+            var rect = request.PresentationSourceBounds.ToPlatformRectangle();
+            rect.Y = view.Bounds.Height - rect.Bottom;
```

Comment:
This is bad code. Several cases cause it to actually not work.

 - `NSWindowStyle.Borderless` makes the `MainWindow` be null
 - the popup is not correctly positioned
 - the title height is not included anywhere